pkg/operator/reporters/incoming/incoming_reporter.go

```diff
@@ -0,0 +1,130 @@
+}
+
+func (c *IncomingReporter) sync(ctx context.Context, syncContext factory.SyncContext) error {
+	client := c.NewBugzillaClient(ctx)
+	slackClient := c.SlackClient(ctx)
+
+	channelReport, assigneeReports, err := Report(ctx, client, syncContext.Recorder(), &c.config)
+	if err != nil {
+		return err
+	}
+	if len(assigneeReports) == 0 {
+		return nil
+	}
+
+	// In 95% cases this will hit the default component assignees.
+	for assignee, bugs := range assigneeReports {
+		message := fmt.Sprintf("%s\n\n> Please set severity/priority on the bug(s) above and assign to a team member.\n", strings.Join(bugs.reports, "\n"))
+		if err := slackClient.MessageEmail(assignee, message); err != nil {
+			syncContext.Recorder().Warningf("DeliveryFailed", "Failed to deliver:\n\n%s\n\n to %q: %v", message, assignee, err)
+			continue
+		}
+		for _, id := range bugs.bugIDs {
+			if err := c.markAsReported(client, id); err != nil {
+				syncContext.Recorder().Warningf("MarkNotifiedFailed", "Failed to mark bug #%d with AssigneeNotified: %v", id, err)
+			}
+		}
+	}
+
+	if err := slackClient.MessageChannel(channelReport); err != nil {
+		syncContext.Recorder().Warningf("DeliveryFailed", "Failed to deliver new bugs: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func NewIncomingReporter(ctx controller.ControllerContext, schedule []string, operatorConfig config.OperatorConfig, recorder events.Recorder) factory.Controller {
+	c := &IncomingReporter{
+		ControllerContext: ctx,
+		config:            operatorConfig,
+	}
+	return factory.New().WithSync(c.sync).ResyncSchedule(schedule...).ToController("IncomingReporter", recorder)
+}
+
+type AssigneeReport struct {
+	bugIDs  []int
+	reports []string
+}
+
+func Report(ctx context.Context, client cache.BugzillaClient, recorder events.Recorder, config *config.OperatorConfig) (string, map[string]AssigneeReport, error) {
+	incomingBugs, err := getIncomingBugsList(client, config)
+	if err != nil {
+		recorder.Warningf("BugSearchFailed", err.Error())
+		return "", nil, err
+	}
+
+	var channelReport []string
+	assigneeReports := map[string]AssigneeReport{}
+
+	for _, bug := range incomingBugs {
+		bugMessage := fmt.Sprintf(":bugzilla: [NEW][%s/%s][%s] <https://bugzilla.redhat.com/show_bug.cgi?id=%d|%s>", bug.Severity, bug.Priority, strings.Join(bug.Component, ","), bug.ID, bug.Summary)
+		channelReport = append(channelReport, "> "+bugMessage)
+		currentReport, ok := assigneeReports[bug.AssignedTo]
+		if ok {
+			currentReport.reports = append(currentReport.reports, bugMessage)
+			currentReport.bugIDs = append(currentReport.bugIDs, bug.ID)
+			continue
+		}
+		assigneeReports[bug.AssignedTo] = AssigneeReport{
+			bugIDs:  []int{bug.ID},
+			reports: []string{bugMessage},
+		}
+	}
+
+	return strings.Join(channelReport, "\n"), assigneeReports, nil
+}
+
+func (c *IncomingReporter) markAsReported(client cache.BugzillaClient, id int) error {
+	return client.UpdateBug(id, bugzilla.BugUpdate{DevWhiteboard: "AssigneeNotified"})
```

Comment:
this blocks us from notifying on ping-pong, right?

Reply:
ping-pong-ping rather :)

Reply:
we could store the time somehow and look through history when the component was changed. If it is changed after that timestamp, ignore the label.

Reply:
@sttts or we can reset the label when component is changed?